public/js/components/card-choice.jsx

```diff
@@ -33,7 +42,7 @@ export default class CardChoice extends Component {
   handleSubmit = (e) => {
     e.preventDefault();
     this.setState({isSubmitting: true});
-    this.props.createSubscription(this.props.productId, this.state.card);
+    this.props.submitHandler(e, this.state.card);
```

Comment:
why does the props function need to receive the event object? I don't see it being used in your implementation here.